src/silicone/database_crunchers/rms_closest.py

```diff
@@ -0,0 +1,228 @@
+    difference.
+
+    .. math::
+        RMS = \\left ( \\frac{1}{n} \\sum_{t=0}^n (E_l(t) - E_l^{d}(t))^2 \\right )^{1/2}
+
+    where :math:`n` is the total number of timesteps in the lead gas' timeseries,
+    :math:`E_l(t)` is the lead gas emissions timeseries and :math:`E_l^d(t)` is a lead
+    gas emissions timeseries in the database.
+    """
+
+    def derive_relationship(self, variable_follower, variable_leaders):
+        """
+        Derive the relationship between two variables from the database.
+
+        Parameters
+        ----------
+        variable_follower : str
+            The variable for which we want to calculate timeseries (e.g.
+            ``"Emissions|C5F12"``).
+
+        variable_leaders : list[str]
+            The variable we want to use in order to infer timeseries of
+            ``variable_follower`` (e.g. ``["Emissions|CO2"]``).
+
+        Returns
+        -------
+        :obj:`func`
+            Function which takes a :obj:`pyam.IamDataFrame` containing
+            ``variable_leaders`` timeseries and returns timeseries for
+            ``variable_follower`` based on the derived relationship between the two.
+            Please see the source code for the exact definition (and docstring) of the
+            returned function.
+
+        Raises
+        ------
+        ValueError
+            ``variable_leaders`` contains more than one variable.
+
+        ValueError
+            There is no data for ``variable_leaders`` or ``variable_follower`` in the
+            database.
+        """
+        self._check_iamdf_lead(variable_leaders)
+        iamdf_follower = self._get_iamdf_section(variable_follower)
+        iamdf_lead = self._db.filter(variable=variable_leaders)
+
+        leader_unit = iamdf_lead["unit"].unique()
+        assert len(leader_unit) == 1  # TODO: use functionality in #13
+        leader_unit = leader_unit[0]
+
+        data_follower_time_col = iamdf_follower.time_col
+
+        def filler(in_iamdf, interpolate=False):
+            """
+            Filler function derived from :obj:`DatabaseCruncherRMSClosest`.
+
+            Parameters
+            ----------
+            in_iamdf : :obj:`pyam.IamDataFrame`
+                Input data to fill data in
+
+            interpolate : bool
+                If the lead timeseries years don't match those in the database, should
+                we interpolate the database timeseries or simply discard them? TODO:
+                test this.
+
+            Returns
+            -------
+            :obj:`pyam.IamDataFrame`
+                Filled in data (without original source data)
+
+            Raises
+            ------
+            ValueError
+                If there are any inconsistencies between the timeseries, units or expectations of the program and the
+                in_amdf, compared to the database passed to the cruncher, one of many errors are thrown.
```

Comment:
```suggestion
                in_iamdf, compared to the database passed to the cruncher.
```

All the errors are ValueErrors, only the message changes so can leave like this.